service/history/transferQueueProcessorBase.go

```diff
@@ -300,7 +302,39 @@ func (t *transferQueueProcessorBase) recordWorkflowClosed(
 		SearchAttributes:   searchAttributes,
 	}
 
-	return t.visibilityMgr.RecordWorkflowExecutionClosed(request)
+	if err := t.visibilityMgr.RecordWorkflowExecutionClosed(request); err != nil {
+		return err
+	}
+
+	clusterConfiguredForArchival := t.shard.GetService().GetArchivalMetadata().GetVisibilityConfig().ClusterConfiguredForArchival()
+	domainConfiguredForArchival := domainEntry.GetConfig().VisibilityArchivalStatus == workflow.ArchivalStatusEnabled
+	if clusterConfiguredForArchival && domainConfiguredForArchival {
+		archiveURI, err := archiver.NewURI(domainEntry.GetConfig().VisibilityArchivalURI)
+		if err != nil {
+			return err
+		}
+		visibilityArchiver, err := t.shard.GetService().GetArchiverProvider().GetVisibilityArchiver(archiveURI.Scheme(), common.HistoryServiceName)
+		if err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), t.shard.GetConfig().TransferProcessorVisibilityArchivalTimeLimit())
+		defer cancel()
+		return visibilityArchiver.Archive(ctx, archiveURI, &archiver.ArchiveVisibilityRequest{
```

Comment:
Emit metric on failure

Reply:
We want to be able to compute a success graph like we do for histories. Make sure we have all the metrics we need to do this.

Reply:
Emitted metrics when visibility archive is attempted or failed. For latency metrics, I think they should be emitted by the archiver itself.